### PR TITLE
feat(auth): enable refresh-token based session handling for portal apps

### LINKED ISSUE #402: [Trader App], [OGA App] Use refresh-token based session handling consistently
## Current Limitation

`portals/apps/trader-app` and `portals/apps/oga-app` rely on default Asgardeo session behavior and are vulnerable to session expiry during longer user workflows. The apps need refresh-token-capable auth configuration and SDK-managed token refresh so users are not logged out mid-operation.

## Suggested Improvement

Update both `portals/apps/trader-app` and `portals/apps/oga-app` to request `offline_access`, ensure the IdP application config issues refresh tokens for the SPA clients, and enable SDK-native periodic token refresh through `AsgardeoProvider` instead of custom tab-level refresh logic. Keep storage explicit and align token validity settings across both portals.

## Version

`ae14498f`

## Additional Context

`trader-app` was investigated first and confirmed that the installed `@asgardeo/react` / `@asgardeo/browser` runtime supports `periodicTokenRefresh`. The same auth/session treatment should be applied to `oga-app` for consistency. Relevant app paths: `portals/apps/trader-app` and `portals/apps/oga-app`.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Enable refresh-token based session handling for the portal applications so longer user workflows are less likely to be interrupted by token expiry. This aligns the frontend auth configuration with the IdP token settings needed for SDK-managed refresh.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- Updated `portals/apps/trader-app/src/main.tsx` to request `offline_access`, make Asgardeo storage explicit, and enable SDK-native `periodicTokenRefresh`
- Updated `portals/apps/oga-app/src/main.tsx` with the same Asgardeo refresh-token and periodic refresh configuration
- Updated `portals/apps/trader-app/.env.example` and `portals/apps/oga-app/.env.example` so the documented default scopes include `offline_access`
- Adjusted `idp/02-sample-resources.sh` token validity settings to align the sample IdP app configuration with the portal refresh-token behavior

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

Local verification performed:
- `pnpm --filter trader-app type-check`
- `pnpm --filter trader-app build`
- `pnpm --filter oga-app type-check`
- `pnpm --filter oga-app build`

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #402

## Screenshots/Demo

Not applicable.

## Additional Notes

The portal apps now rely on the Asgardeo SDK's built-in `periodicTokenRefresh` rather than custom refresh logic. Refresh-token issuance still needs to be enabled in the IdP configuration for the relevant SPA clients.

## Deployment Notes

Ensure the IdP applications used by `trader-app` and each `oga-app` instance are configured to issue refresh tokens and allow the `offline_access` scope.

### CODE REVIEW

File: portals/apps/trader-app/src/main.tsx [hunk 1]
```diff
@@ -17,32 +22,34 @@ const normalizeIdpPlatform = (value: string): 'AsgardeoV2' | 'Asgardeo' | 'Ident
...
-const IDP_PLATFORM = normalizeIdpPlatform(getEnv('VITE_IDP_PLATFORM', 'AsgardeoV2')!)
+const APP_URL = getEnv('VITE_APP_URL', window.location.origin)
+const CLIENT_ID = getEnv('VITE_IDP_CLIENT_ID', 'TRADER_PORTAL_APP')
+const IDP_BASE_URL = getEnv('VITE_IDP_BASE_URL', 'https://localhost:8090')
+const IDP_PLATFORM = normalizeIdpPlatform(getEnv('VITE_IDP_PLATFORM', 'AsgardeoV2'))
 const rawScopes = getEnv('VITE_IDP_SCOPES')
 const IDP_SCOPES = rawScopes
   ? rawScopes.split(',').map((s: string) => s.trim())
-  : ["openid", "profile", "email"]
+  : ["openid", "profile", "email", "offline_access"]
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portals/apps/trader-app/src/main.tsx]: Added "groups"

**Review by ginaxu1 — DISMISSED** [review]

**Review by sthanikan2000 — CHANGES_REQUESTED** [review]
I set the `access_token` validity period as 30 seconds and `Reload` my browser app. Until then the `token` was refreshing, but after that it was not refreshing token. Can you check whether the page reload erase the `refresh_token` from the session storage?

File: portals/apps/oga-app/src/main.tsx [hunk 2]
```diff
@@ -23,23 +29,25 @@ const IDP_PLATFORM = normalizeIdpPlatform(getEnv('VITE_IDP_PLATFORM', 'AsgardeoV
 const rawScopes = getEnv('VITE_IDP_SCOPES')
 const IDP_SCOPES = rawScopes
   ? rawScopes.split(',').map((scope: string) => scope.trim())
-  : ['openid', 'profile', 'email']
+  : ['openid', 'profile', 'email', 'offline_access']
```
Comment (sthanikan2000) [inline portals/apps/oga-app/src/main.tsx]: `refresh_token` is returning by the current login flow. Seems like we don't need this scope. Can you check whether this configuration is really needed?

**Review by sthanikan2000 — CHANGES_REQUESTED** [review]

File: portals/apps/oga-app/src/main.tsx (hunk 2 above)
Comment (mushrafmim) [reply portals/apps/oga-app/src/main.tsx]: Removed `offline_access`, it is used to request tokens in the background, even when the user is not logged in. I think usually used in mobile apps as background tasks.

**Comment by mushrafmim** [comment]
> I set the `access_token` validity period as 30 seconds and `Reload` my browser app. Until then the `token` was refreshing, but after that it was not refreshing token. Can you check whether the page reload erase the `refresh_token` from the session storage?

The problem is, when the browser is refreshed, it stops token refresh process. After that when a request is made and 401 is returned, the default http client from the Asgardeo, retries to obtain a token using the refresh token. Since we are not using the default http client of Asgardeo, we get rid of the access token.

To mitigate the problem temporarily, have reverted the expiry time of the token to 1 hour. After bringing proper token refresh process we can fix this issue.

Have added a refresh button in the PR #413, which will be enabling us to get rid of browser refresh in the app.

**Review by sthanikan2000 — APPROVED** [review]
LGTM!

Created the issue for the bug I found #418

**Review by ginaxu1 — APPROVED** [review]
